nncf/tensorflow/pruning/filter_pruning/algorithm.py

```diff
@@ -520,25 +470,23 @@ def _calculate_filters_importance_in_group(self, group: Cluster[PrunedLayerInfo]
         return cumulative_filters_importance
 
     def _update_benchmark_statistics(self):
-        tmp_in_channels, tmp_out_channels = calculate_in_out_channels_by_masks(
-                pruning_groups=self._pruned_layer_groups_info.get_all_clusters(),
-                num_of_sparse_elements_by_node=self._calculate_num_of_sparse_elements_by_node(),
-                full_input_channels=self._layers_in_channels,
-                full_output_channels=self._layers_out_channels,
-                pruning_groups_next_nodes=self._next_nodes)
-
-        self.current_filters_num = count_filters_num(self._original_graph,
-                                                     op_metatypes=GENERAL_CONV_LAYER_METATYPES +
-                                                     LINEAR_LAYER_METATYPES,
-                                                     output_channels=tmp_out_channels)
+        tmp_in_channels, tmp_out_channels = \
+            self._shape_pruning_proc.calculate_in_out_channels_by_masks(
```

Comment:
`calculate_in_out_channels_by_masks` calculate something that is needed for calculation of flops via `weights_flops_calc` only. Why not hide this calculation in the `weight_flops_calc` then, or maybe in another entity?

Unfortunately, `ShapePruninigProcessor` and `WeightsFlopsCalculator` don't encapsulate a lot - just metatypes and pruning types. 
It would be super-great to have a class with a simple and minimal interface, like `calc_flops` and `set_graph`.

Reply:
1) I made separate entities because elastic width needs only `ShapePruninigProcessor` and multi elasticity handler needs only `WeightsFlopsCalculator`. Maybe it worth to have one new entity when you need both classes which compose `ShapePruninigProcessor`  and `WeightsFlopsCalculator`, let's discuss it.

2) Before I dive into NAS algo the classes kept all needed information inside. But when I found that NAS can change not just graph, but pruning groups, I decided to keep only things which will not be changed during any algorithm. 

There are two options: keep params and than update it or feed params and explicitly ask algorithm to manage such parameters. Param feeding is more explicit way, so I vote for it

Reply:
Other thing I was thinking is that `next_nodes` and `pruning_groups` always should be in one place, this is thoughts for future common pruning algo